packages/react/src/components/Provider/ImagePreloader.tsx

```diff
@@ -0,0 +1,27 @@
+import { useEffect, useState } from 'react'
+import { useFrigade } from '@/hooks/useFrigade'
+
+export function ImagePreloader() {
+  const [hasPreloaded, setHasPreloaded] = useState(false)
+  const { frigade } = useFrigade()
+
+  useEffect(() => {
+    ;(async () => {
+      if (hasPreloaded) {
```

Comment:
IIRC `hasPreloaded` shouldn't be needed since the effect has an empty deps array and will only run once. If it appears to be running twice that's just React dev mode being a precocious lil gremlin about it.

Reply:
You are correct sir!